## Домашнее задание №5
***

In [5]:
import numpy as np

### Задача 1
Когда используется критерий Стьюдента, а когда Z –критерий? 

***
### Ответ:
Z-критерий и t-критерий Стьюдента являются параметрическими критериями, Они предполагают, что выборка взята из нормально распределенной генеральной совокупности и дисперсии в группах равны. \
Z-критерий используется, когда известна $\sigma$ генеральной совокупности,\
если размер выборки < 30 и  $\sigma$ генеральной совокупности неизвестна, то используется t-критерий Стьюдента.
***

### Задача 2
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

#### Решение:
1. Сформулируем нулевую и альтернативную гипотезы:\
$H0:$ **Статистически значимых различий между генеральной совокупностью и выборкой нет** $\mu=\mu_0$,\
где $\mu$ - среднее значение выборки, $\mu_0$ среднее значение генеральной совокупности.  
$H1:$ **Статистически значимые различия между генеральной совокупностью и выборкой есть** $\mu>\mu_0$  
2. Уровень статистической значимости $\alpha=0.05$ 
3. Поскольку нам известна дисперсия, то для тестирования гипотезы будем использовать Z-критерий.\
Рассчитаем наблюдаемое(расчетное) значение Z-критерия по формуле:
$z_н=\frac{\bar X-\mu_0}{\sigma/\sqrt n}$, где\
$\bar X=\mu=17.5$ среднее по выборке,  
$\sigma=\sqrt 4=2$ среднее квадратическое отклонение генеральной совокупности,  
$n=100$ размер выборки.  


In [6]:
z_n = (17.5 - 17)/(4**0.5/100**0.5)
print(f'Наблюдаемое значение Z-критерия = {z_n: .2f}')

Наблюдаемое значение Z-критерия =  2.50


4. Найдем табличное значение $Z_t$-критерия, для $\alpha=0.05$\
соответствующие значения вероятности можно посмотреть на этом [ресурсе](https://datascience.eu/ru/%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0-%D0%B8-%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D0%BA%D0%B0/z-%D1%82%D0%B0%D0%B1%D0%BB%D0%B8%D1%86%D0%B0/)\
Табличное значение Z-критерия = 1,65


5. Сравним расчетное значение с табличным:\
$z_н>z_t$, расчетное значение Z попадает в область принятия альтернативной гипотезы, следовательно отвергаем нулевую гипотезу и принимаем альтернативную гипотезу.

***
### Ответ:
Принимаем альтернативную гипотезу $H1:$ **Статистически значимые различия между генеральной совокупностью и выборкой обнаружены**: $\bar X=\mu=17.5$ при уровене статистической значимости $\alpha=0.05$ 
***

### Задача 3
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

#### Решение:
1. Сформулируем нулевую и альтернативную гипотезы:\
$H0:$ **Статистически значимых различий между генеральной совокупностью и выборкой нет** $\mu=\mu_0=200$,\
где $\mu$ - среднее значение выборки, $\mu_0$ среднее значение генеральной совокупности.  
$H1:$ **Статистически значимые различия между генеральной совокупностью и выборкой есть** $ \mu\neq\mu_0$ (двусторонний тест)  
2. Для двустороннего теста заданный уровень статистической значимости разделим пополам: $\alpha=0.01/2=0.005$ 
3. Поскольку нам не известна дисперсия генеральной совокупности, то для тестирования гипотезы будем использовать t-критерий Стьюдента.\
Рассчитаем наблюдаемое(расчетное) значение t-критерия Стьюдента по формуле:
$t_н=\frac{\bar X-\mu_0}{\sigma_н/\sqrt n}$\
Для двустороннего теста гипотезы достаточно найти только одно значение из $ t_{\alpha/2}$ и  $-t_{\alpha/2}$, т.к. распределение симметрично

In [7]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Рассчитаем среднее арифметическое выборки

In [8]:
def Get_mean(arr):
    return (sum(arr)/(len(arr)))

In [9]:
print(f'Среднее значение по выборке = {Get_mean(x): .2f}')

Среднее значение по выборке =  198.50


Рассчитаем среднее квадратическое отклонение для выборки

In [10]:
def Get_deviation(arr):
    square_dev=(arr-Get_mean(arr))**2
    return (sum(square_dev)/(len(arr)-1))**(1/2)

In [11]:
print(f'Среднее квадратичное отклонение для выборки = {Get_deviation(x): .4f}')

Среднее квадратичное отклонение для выборки =  4.4535


Рассчитаем наблюдаемое значение $t_n$ - критерия Стюдента

In [12]:
t_n = abs(Get_mean(x) - 200)/((Get_deviation(x))/10**0.5)

In [13]:
print(f'Расчетное значение t-критерия Стьюдента = {t_n: .4f}')

Расчетное значение t-критерия Стьюдента =  1.0651


Табличное значение $t$-критерия Стьюдента можно посмотреть на этом [ресурсе](https://www.matematicus.ru/teoriya-veroyatnosti/tablitsy/tablitsa-znachenij-kriteriya-styudenta-t-kriteriya/)\
$t_t$ = 3.6897\
Поскольку $|t_н|<t_t$, **принимаем** нулевую гипотезу на уровне значимости $\alpha=0.01$

***
### Ответ:
Принимаем нулевую гипотезу $H0:$ **Статистически значимых различий между генеральной совокупностью и выборкой не обнаружены**: $\bar X=\mu=200$ при уровене статистической значимости $\alpha=0.01$ 
***

### Задача 4
Есть ли статистически значимые различия в росте дочерей?\
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165\
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163\
*Задачу 4 решать с помощью функции*

#### Решение
Исходя из условия задачи требуется провести двухвыборочный тест с зависимыми выборками.
1. Сформулируем нулевую и альтернативную гипотезы:\
$H0:$ **Статистически значимых различий в росте матерей и дочерей нет**\
$H1:$ **Статистически значимые различия в росте матерей и дочерей есть**
2. Уровень статистической значимости по умолчанию $\alpha=0.05$ 

In [14]:
import scipy.stats as stats

In [26]:
x = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])

In [27]:
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])

In [28]:
stats.ttest_rel(x, y)

TtestResult(statistic=0.6648478531431979, pvalue=0.5228168632983574, df=9)

Поскольку рассчитанное p-значение $p-value=0.52>\alpha=0.05$,то принимаем нулевую гипотезу. 

***
### Ответ:
Поскульку $P-value$ больше уровня значимости $\alpha$, принимаем нулевую гипотезу\
$H0:$ **Статистически значимых различий в росте матерей и дочерей нет**